lib/galaxy/webapps/galaxy/api/lda_datasets.py

```diff
@@ -194,17 +195,17 @@ def _get_current_roles( self, trans, library_dataset):
         modify_roles = set( trans.app.security_agent.get_roles_for_action( library_dataset, trans.app.security_agent.permitted_actions.LIBRARY_MODIFY ) )
         manage_roles = set( dataset.get_manage_permissions_roles( trans ) )
 
-        access_dataset_role_list = [ access_role.name for access_role in access_roles ]
-        manage_dataset_role_list = [ manage_role.name for manage_role in manage_roles ]
-        modify_item_role_list = [ modify_role.name for modify_role in modify_roles ]
+        access_dataset_role_list = [ ( access_role.name, trans.security.encode_id( access_role.id ) ) for access_role in access_roles ]
+        manage_dataset_role_list = [ ( manage_role.name, trans.security.encode_id( manage_role.id ) ) for manage_role in manage_roles ]
+        modify_item_role_list = [ ( modify_role.name, trans.security.encode_id( modify_role.id ) ) for modify_role in modify_roles ]
```

Comment:
I think this will change the output of the following API calls:
- GET /api/libraries/datasets/{encoded_dataset_id}/permissions
- POST /api/libraries/datasets/{encoded_dataset_id}/permissions

They are not available through BioBlend yet, but this should probably be highlighted in the Release Notes.